pkg/gatherers/conditional/conditional_gatherer.go

```diff
@@ -172,6 +172,57 @@ var defaultGatheringRules = []GatheringRule{
 			},
 		},
 	},
+	{
+		Conditions: []ConditionWithParams{
+			{
+				Type: AlertIsFiring,
+				Alert: &AlertConditionParams{
+					Name: "AlertmanagerFailedReload",
+				},
+			},
+		},
+		GatheringFunctions: GatheringFunctions{
+			GatherContainersLogs: GatherContainersLogsParams{
+				AlertName: "AlertmanagerFailedReload",
+				Container: "alertmanager",
+				TailLines: 50,
+			},
+		},
+	},
+	{
+		Conditions: []ConditionWithParams{
+			{
+				Type: AlertIsFiring,
+				Alert: &AlertConditionParams{
+					Name: "PrometheusTargetSyncFailure",
+				},
+			},
+		},
+		GatheringFunctions: GatheringFunctions{
+			GatherContainersLogs: GatherContainersLogsParams{
+				AlertName: "PrometheusTargetSyncFailure",
+				Container: "prometheus",
+				TailLines: 50,
+			},
+		},
+	},
+	{
+		Conditions: []ConditionWithParams{
+			{
+				Type: AlertIsFiring,
+				Alert: &AlertConditionParams{
+					Name: "ThanosRuleQueueIsDroppingAlerts",
+				},
+			},
+		},
+		GatheringFunctions: GatheringFunctions{
+			GatherContainersLogs: GatherContainersLogsParams{
+				AlertName: "ThanosRuleQueueIsDroppingAlerts",
```

Comment:
I am wondering where is this alert defined please? I was looking at `thanos-querier` `PrometheusRule` in my 4.10-nightly cluster and I didn't see this alert within the defintion.

Reply:
And related question is where can I find the `thanos-ruler` container please?

Reply:
Thanos ruler is only created when you enable UWM.
https://docs.openshift.com/container-platform/4.9/monitoring/monitoring-overview.html
https://docs.openshift.com/container-platform/4.9/monitoring/enabling-monitoring-for-user-defined-projects.html#enabling-monitoring-for-user-defined-projects_enabling-monitoring-for-user-defined-projects

Reply:
This is where it is defined.
https://github.com/openshift/cluster-monitoring-operator/blob/master/assets/thanos-ruler/thanos-ruler-prometheus-rule.yaml

Reply:
Thank you